### ISSUE TITLE
using 'lessThan' rule in conjunction with 'max' supresses 'max' validation

### ISSUE THREAD

**Issue by levi2ki** [jquense/yup#2263]
**Describe the bug**
When using `lessThan` rule in conjunction with `max` rule and they both failed, only `lessThan` error will be thrown and `max` error will be omitted despite { abortEarly: false }

**To Reproduce**
https://codesandbox.io/p/sandbox/yup-test-case-gg1g1

**Expected behavior**
Expected to receive both errors within validation 'from' attribute

**Platform (please complete the following information):**
 - Browser chrome
 - Version 131.0.6778.205

**Additional context**
I checked some combinations like:
1. lessThan and min - green test - both errors received
2. moreThan and max - green test - both errors received
3. moreThan and min - green test - both errors received

**Comment by jquense** [comment]
yes because the rules are mutually exclusive, it doesn't sense to say something must be less than 5 and max 10, that is just "less than 5"
